src/simulations/collect.jl

```diff
@@ -164,6 +165,121 @@ function run!(
     return df_agent, df_model
 end
 
+"""
+    offline_run!(model, agent_step! [, model_step!], n::Integer; kwargs...) → agent_df, model_df
+    offline_run!(model, agent_step!, model_step!, n::Function; kwargs...) → agent_df, model_df
+
+Run the model while dumping data to CSV files. See [`run!`](@ref) for details.
+
```

Comment:
I think you can say more here. A lengthier description can remove ambiguity. I suggest:
```
Do the same as [`run`](@ref), but write output directly to a CSV file instead of
collecting it into an in-memory dataframe. Useful in scenarios where the amount
of collected data is expected to exceed available memory.
```